plugins/woocommerce/includes/class-wc-query.php

```diff
@@ -587,7 +587,10 @@ public function get_catalog_ordering_args( $orderby = '', $order = '' ) {
 		// Get ordering from query string unless defined.
 		if ( ! $orderby ) {
 			// phpcs:ignore WordPress.Security.NonceVerification.Recommended
-			$orderby_value = isset( $_GET['orderby'] ) ? wc_clean( (string) wp_unslash( $_GET['orderby'] ) ) : wc_clean( get_query_var( 'orderby' ) );
+			$orderby_value = isset( $_GET['orderby'] ) ? wc_clean( wp_unslash( $_GET['orderby'] ) ) : wc_clean( get_query_var( 'orderby' ) );
+			if ( is_array( $orderby_value ) ) {
+				$orderby_value = $orderby_value[0];
+			}
```

Comment:
🤔 I'm wondering if this is necessary, or if it might even cause a problem, given that [further down](https://github.com/woocommerce/woocommerce/blob/8309364c4f0ac56dda0f42047d91895960c5f9f7/plugins/woocommerce/includes/class-wc-query.php#L601) it expects that `$orderby_value` _could_ be an array, and the second item in the array might actually be an `order` value (`asc` or `desc`).

I don't actually know if this would cause a problem, just thinking of ways this could come back to bite us later...

Reply:
Good point 🤔 
I think it is necessary but only when we are dealing with orderby coming from `$_GET`, otherwise cases like the one in the `test_get_catalog_ordering_args_when_GET_is_array` will fail.
I've changed it here https://github.com/woocommerce/woocommerce/pull/55586/commits/6ad5d2f1b31b5309d7339a13942f5571103de54e to only getting the first element of the array on that case, wdyt?

Reply:
👍 Sounds good to me